docker-compose.yml

```diff
@@ -42,40 +42,30 @@ services:
       - minio
 
   starrocks-fe:
-    image: starrocks/fe-ubuntu:3.2-latest
+    build:
+      context: .
+      dockerfile: ./DockerFile_FrontEnd
     hostname: starrocks-fe
     container_name: starrocks-fe
     user: root
-    command:
-      - /bin/bash
-      - -c
-      - |
-        echo "# enable shared data, set storage type, set endpoint" >> /opt/starrocks/fe/conf/fe.conf
-        echo "run_mode = shared_data" >> /opt/starrocks/fe/conf/fe.conf
-        echo "cloud_native_storage_type = S3" >> /opt/starrocks/fe/conf/fe.conf
-        echo "aws_s3_endpoint = minio:9000" >> /opt/starrocks/fe/conf/fe.conf
-
-        echo "# set the path in MinIO" >> /opt/starrocks/fe/conf/fe.conf
-        echo "aws_s3_path = starrocks" >> /opt/starrocks/fe/conf/fe.conf
-
-        echo "# credentials for MinIO object read/write" >> /opt/starrocks/fe/conf/fe.conf
-        echo "aws_s3_access_key = AAAAAAAAAAAAAAAAAAAA" >> /opt/starrocks/fe/conf/fe.conf
-        echo "aws_s3_secret_key = BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB" >> /opt/starrocks/fe/conf/fe.conf
-        echo "aws_s3_use_instance_profile = false" >> /opt/starrocks/fe/conf/fe.conf
-        echo "aws_s3_use_aws_sdk_default_behavior = false" >> /opt/starrocks/fe/conf/fe.conf
-
-        echo "# Set this to false if you do not want default" >> /opt/starrocks/fe/conf/fe.conf
-        echo "# storage created in the object storage using" >> /opt/starrocks/fe/conf/fe.conf
-        echo "# the details provided above" >> /opt/starrocks/fe/conf/fe.conf
-        echo "enable_load_volume_from_conf = true" >> /opt/starrocks/fe/conf/fe.conf
-
-        /opt/starrocks/fe/bin/start_fe.sh
+    environment:
+      - STARROCKS_HOME=/starrocks/fe1
+      - HTTP_PORT=8031
+      - THRIFT_RPC_PORT=9021
+      - SQL_QUERY_PORT=9031
+      - EDIT_LOG_PORT=9011
```

Comment:
why not just using the default ports assigned by the entrypoint?

Reply:
Or should we not assign default ports for those in the entrypoint.sh and exit if they are not set?

Reply:
This is just an example for how to specify the ports. If you want default ports, delete / comment out these lines